### ISSUE TITLE
Can't disable shellcheck globally for a file

### ISSUE THREAD

**Issue by 64kramsystem** [koalaman/shellcheck#2252]
#### For bugs

- SC1073
- 0.7.1
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

# shellcheck disable
```

#### Here's what shellcheck currently says:

```
Line 3:
# shellcheck disable
^-- SC1073: Couldn't parse this shellcheck directive. Fix to allow more checks.
                    ^-- SC1072: Expected '=' after directive key. Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:

According to #777, it should be possible to globally disable Shellcheck for a file:

> [...] In 0.4.4 and later, adding a disable directive at the top of the script (before any commands) works exactly as you suggest.

However, it's not clear how this works (or if it does). I've checked the [wiki page](https://github.com/koalaman/shellcheck/wiki/Directive), but I don't see any way to perform a global disable, as mentioned in the issue.

I've created a separate issue because I can't reopen the original one.

**Comment by brother** [comment]
You might have intepreted the issue #777 a tad bit wrong. It's about disabling checking for an error for the entire file not for the next line or closure as is the normal behaviour.

````
brother ~/git/other/gleshys (main *%)$ head -n2 attic/functions
#!/bin/bash

brother ~/git/other/gleshys (main *%)$ shellcheck -f gcc attic/functions 
attic/functions:23:9: note: Double quote to prevent globbing and word splitting. [SC2086]
attic/functions:27:6: note: $/${} is unnecessary on arithmetic variables. [SC2004]
attic/functions:35:3: note: read without -r will mangle backslashes. [SC2162]
brother ~/git/other/gleshys (main *%)$ head -n2 attic/functions
#!/bin/bash
# shellcheck disable=SC2004
brother ~/git/other/gleshys (main *%)$ shellcheck -f gcc attic/functions 
attic/functions:23:9: note: Double quote to prevent globbing and word splitting. [SC2086]
attic/functions:35:3: note: read without -r will mangle backslashes. [SC2162]
```

Disabling ALL checks for the file is not possible, maybe you could avoid executing shellcheck with the problematic (?) file of yours? 

Maybe only search for errors and not all severities? (`-S error`)

**Comment by saveriomiroddi** [comment]
> You might have intepreted the issue #777 a tad bit wrong. It's about disabling checking for an error for the entire file not for the next line or closure as is the normal behaviour.
> 
> ```
> brother ~/git/other/gleshys (main *%)$ head -n2 attic/functions
> #!/bin/bash
> 
> brother ~/git/other/gleshys (main *%)$ shellcheck -f gcc attic/functions 
> attic/functions:23:9: note: Double quote to prevent globbing and word splitting. [SC2086]
> attic/functions:27:6: note: $/${} is unnecessary on arithmetic variables. [SC2004]
> attic/functions:35:3: note: read without -r will mangle backslashes. [SC2162]
> brother ~/git/other/gleshys (main *%)$ head -n2 attic/functions
> #!/bin/bash
> # shellcheck disable=SC2004
> brother ~/git/other/gleshys (main *%)$ shellcheck -f gcc attic/functions 
> attic/functions:23:9: note: Double quote to prevent globbing and word splitting. [SC2086]
> attic/functions:35:3: note: read without -r will mangle backslashes. [SC2162]
> ```
> 
> Disabling ALL checks for the file is not possible, maybe you could avoid executing shellcheck with the problematic (?) file of yours?
> 
> Maybe only search for errors and not all severities? (`-S error`)

Ahhh... sorry, I misunderstood big time :grin:

The issue is actually very similar to that user - I have a plugin that automatically invoke Shellcheck, and it's more convenient to have an explicit directive in the shell file, than configuring the plugin (I also originally thought that disabling all checks was an intended feature).

Since this is just a convenience and it was not intended, though, it'd be a feature request :grin: 

However, since this is not in the project plans, feel free to close. Great work :smile:

**Comment by bland328** [comment]
Piggybacking here _(I'll open a fresh issue, if preferred)_ to suggest/request a mechanism to entirely disable shellcheck for a given file.

My use case is that I have many reference files in a VS Code project that are littered with (legitimate) shellcheck complaints that I'd like to silence; since I'm using an IDE+shellcheck plugin, it is challenging/impractical to selectively silence these files any other way without also losing syntax highlighting.

Optimally, I'd like to see something like:
```bash
# shellcheck disable
```
or
```bash
# shellcheck disable=*
```
simply silence shellcheck for an entire file.

**Comment by koalaman** [comment]
Since 0.7.2 you've been able to specify a range, so you can technically use:

```
#!/bin/bash
# shellcheck disable=SC1000-SC9999
```

There should really be a better way though.

**Comment by bland328** [comment]
@koalaman, thanks very much for that, though unfortunately I've reverted to 0.7.1 for now, pending a fix for the https://github.com/koalaman/shellcheck/issues/2217 regression.

**Comment by saveriomiroddi** [comment]
> Since 0.7.2 you've been able to specify a range, so you can technically use:
> 
> ```
> #!/bin/bash
> # shellcheck disable=SC1000-SC9999
> ```
> 
> There should really be a better way though.

Excellent. Issue can be closed, for me :)

**Comment by misaki-web** [comment]
Note that since shellcheck v0.8.0, there's a [shortcut to disable all directives](https://github.com/koalaman/shellcheck/commit/09aa15c9b7bc33135178c7fb3338a870f7ce6a88):

```bash
# shellcheck disable=all
```

**Comment by qneill-sifive** [comment]
> Note that since shellcheck v0.8.0, there's a [shortcut to disable all directives](https://github.com/koalaman/shellcheck/commit/09aa15c9b7bc33135178c7fb3338a870f7ce6a88):
> 
> ```shell
> # shellcheck disable=all
> ```

Drive-by comment: With `# shellcheck disable=all` it seems this issue can be closed?
